components/base/wheeled/wheeled_base.go

```diff
@@ -115,6 +133,96 @@ func (base *wheeledBase) Spin(ctx context.Context, angleDeg, degsPerSec float64,
 	rpm, revolutions := base.spinMath(angleDeg, degsPerSec)
 
 	return base.runAll(ctx, -rpm, revolutions, rpm, revolutions)
+
+}
+
+func (base *wheeledBase) spinWithMovementSensor(ctx context.Context, angleDeg, degsPerSec float64, extra map[string]interface{}) error {
+	wheelrpm, revs := base.spinMath(angleDeg, degsPerSec)
+
+	currYaw, err := getCurrentYaw(ctx, base.orientationSensor, extra) // from -179 to 179
+	if err != nil {
+		return err
+	}
+
+	errCounter := 0
+	targetYaw := calculatedDomainLimitedAngleError(angleDeg, currYaw)
+	timer := time.NewTicker(time.Duration(pollTimeMs * float64(time.Millisecond))) //~rename
+	for {
+
+		select {
+		case <-ctx.Done():
+			return nil
+		default:
+		}
+		select {
```

Comment:
Looking into this myself since it'll be relevant to a lot of control loop work. ticker ensures that the loop runs at exactly the time interval you want, i.e. each 100ms we start this loop again. While sleep needs all the code to execute, then adds a wait. 
https://stackoverflow.com/questions/53315716/which-one-of-the-two-is-idiomatic-way-time-sleep-or-ticker

Some more differentiation between the two:
https://levelup.gitconnected.com/timer-and-ticker-using-golang-933a6d00a832